docs/src/main/asciidoc/config-your-quarkus-applications-using-a-yaml-file_howto.adoc

```diff
@@ -0,0 +1,407 @@
+  quarkus:
+    datasource:
+      url: jdbc:postgresql://localhost:5432/some-database
+      driver: org.postgresql.Driver
+      username: quarkus
+      password: quarkus
+----
+
+// include::modules/con_property-expressions.adoc[leveloffset=+1]
+[id="con_property-expressions"]
+== Property expressions
+
+Property expressions are combinations of property references and plain text strings that you can use to substitute values of properties in your configuration.
+
+Much like a variable, you can use a property expression in {project-name} to substitute a configuration property value instead of hardcoding it.
+A property expression is resolved when `java.util.Properties` reads the property's value from a configuration source in your application.
+
+If a configuration property is read from your configuration at compile time, the property expression is also resolved at compile time.
+If the configuration property is overridden at runtime, its value is resolved at runtime.
+
+You can resolve property expressions by using more than one configuration source.
+This means that you can use the value of a property in one configuration source to expand a property expression that you use in another.
+
+If the value of a property in an expression cannot be resolved, and you do not set a default value for the expression, your application encounters a `NoSuchElementException`.
+
+//Property expressions use the `${<property_name>}` syntax.
+
+// include::modules/ref_example-usage-of-property-expressions-using-a-yaml-file.adoc[leveloffset=+2]
+
+[id="ref_example-usage-of-property-expressions-using-a-yaml-file"]
+== Example usage of property expressions using a YAML file
+
+This section shows examples of using property expressions to achieve flexibility when configuring your {project-name} application.
+
+.Example `application.yaml` file
+[source,properties,options="nowrap"]
+----
+mach: 3
+x:
+  factor: 2.23694
+
+display:
+  mach: ${mach}
+  unit:
+    name: "mph"
+    factor: ${x.factor}
+----
+
+[NOTE]
+====
+You can reference nested properties by using the `.` (dot) separator, as in `{x.factor}`.
+====
+
+.Additional resources
+
+* For more information about property expressions, see xref:con_property-expressions[Property expressions].
+// * For an example of property expressions using a properties file, see link:{URL_CONFIGURATION_QUARKUS}#ref_example-usage-of-property-expressions_quarkus-configuration-guide[Example usage of property expressions]. // TBD update and enable this ^ link later when this guide becomes available
+
+// include::modules/con_external-application-yaml-file-for-configuring-properties-at-runtime.adoc[leveloffset=+1]
+[id="con_external-application-yaml-file-for-configuring-properties-at-runtime"]
+== External application.yaml file for configuring properties at runtime
+
+To configure your application properties at runtime, add your `application.yaml` file to the `config` directory.
```

Comment:
`application.properties` mention.

Reply:
Use a "application configuration" not "properties"

Reply:
Updated. Thanks.